src/_P101_WakeOnLan.ino

```diff
@@ -0,0 +1,475 @@
+        // msgStr += getCustomTaskSettingsError(1); // Report string too long.
+        // errorStr += msgStr;
+        // msgStr    = wolStr + msgStr;
+        // addLog(LOG_LEVEL_INFO, msgStr);
+      }
+
+      if (strlen(macString) == 0) { // MAC Address missing, use default value.
+        strcpy_P(macString, (char *)(F(MAC_STR_DEF_P101)));
+
+        msgStr    = F("MAC Address Not Provided, Populated with Zero Values. ");
+        errorStr += msgStr;
+        addLog(LOG_LEVEL_INFO, wolStr + msgStr);
+      }
+      else if (!validateMac(macString)) { // Suspicious MAC Address. Leave as-is, but warn User.
+        msgStr    = F("ERROR, MAC Address Invalid. ");
+        errorStr += msgStr;
+        msgStr    = wolStr + msgStr;
+        msgStr   += F("[");
+        msgStr   += macString;
+        msgStr   += F("]");
+        addLog(LOG_LEVEL_INFO, msgStr);
+      }
+
+      // Save the user's IP and MAC Address parameters into Custom Settings.
+      safe_strncpy(deviceTemplate[0], ipString,   IP_BUFF_SIZE_P101);
+      safe_strncpy(deviceTemplate[1], macString, MAC_BUFF_SIZE_P101);
+
+      if (errorStr.length() > 0) { // Send error messages (if any) to webform.
+        addHtmlError(errorStr);
+      }
+
+      // Save all the Task parameters.
+      SaveCustomTaskSettings(event->TaskIndex, (byte *)&deviceTemplate, sizeof(deviceTemplate));
+      UDP_PORT_P101 = getFormItemInt(F(FORM_PORT_P101));
+      success       = true;
+      break;
+    }
+
+    case PLUGIN_INIT: {
+      success = true;
+      break;
+    }
+
+    case PLUGIN_READ: {
+      success = false;
+      break;
+    }
+
+    case PLUGIN_WRITE: {
+      bool   taskEnable  = true;
+      byte   parse_error = false;
+      String tmpString   = string;
+      String strings[2];
+
+      //  addLog(LOG_LEVEL_INFO, String(F("--> WOL taskIndex= ")) + String(event->TaskIndex)); // Debug
+
+      //  Warning, event->TaskIndex is incorrectly set in PLUGIN_WRITE if plugin is disabled.
+      //  So checking the task enable status needs special attention.
+      //  See https://github.com/letscontrolit/ESPEasy/issues/3317
+      if (event->TaskIndex < TASKS_MAX) {
+        LoadTaskSettings(event->TaskIndex);
+        taskEnable = Settings.TaskDeviceEnabled[event->TaskIndex];
+        LoadCustomTaskSettings(event->TaskIndex, strings, 2, CUSTOMTASK_STR_SIZE_P101);
+        safe_strncpy(ipString,  strings[0],  IP_BUFF_SIZE_P101);
+        safe_strncpy(macString, strings[1], MAC_BUFF_SIZE_P101);
+      }
+
+      String cmd = parseString(tmpString, 1);
+
+      //  Warning, event->TaskIndex is incorrectly set in PLUGIN_WRITE if plugin is disabled.
+      //  So checking the Device Name needs special attention.
+      //  See https://github.com/letscontrolit/ESPEasy/issues/3317
+      if (cmd.equalsIgnoreCase(F(CMD_NAME_P101)) ||
+          ((event->TaskIndex < TASKS_MAX) && cmd.equalsIgnoreCase(getTaskDeviceName(event->TaskIndex)))) {
```

Comment:
`validTaskIndex`

Reply:
The plugin purposely allows using the task name as the command. But you are correct, someone could very likely name their WOL task as REBOOT.  I will add a test in PLUGIN_WEBFORM_SAVE that prevents using REBOOT as the task name.

Reply:
Maybe also 'reset' :)

I guess we should add (later) a quick check to see if a word is a reserved keyword, based on the included plugins.
This could also speed-up command processing quite a bit.

Related PR: https://github.com/letscontrolit/ESPEasy/pull/3198

Reply:
Got it. I'll include RESET in the reserved naming test.

Reply:
Updated File, see safeName() function, Line 419.
- Thomas